tools/block-generator/runner/runner.go

```diff
@@ -35,17 +35,21 @@ func init() {
 		Use:   "runner",
 		Short: "Run test suite and collect results.",
 		Long:  "Run an automated test suite using the block-generator daemon and a provided conduit binary. Results are captured to a specified output directory.",
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error{
 			fmt.Printf("starting block-generator runner with args: %+v\n", runnerArgs)
-			if err := Run(runnerArgs); err != nil {
-				fmt.Println(err)
+
+			if runnerArgs.Template == "postgres-exporter" && runnerArgs.PostgresConnectionString == "" {
+				return fmt.Errorf("exporting to postgres requires that postgres-connection-string to be set")
```

Comment:
I don't think error handling should happen at this level. In order to make it easier to support generic templates we should just let the Conduit process fail to launch.

In the future we can add better error detection to `startConduit`.

Reply:
This resulted from trying to stay as close to the original `RunnerCmd.MarkFlagRequired("postgres-connection-string")` validation that I had removed. I'm not sure that `startConduit` is really the place to validate this as it currently only has access to parameters `dataDir, conduitBinary` and `round`.